src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -80,8 +80,11 @@ public async Task<LSP.CompletionItem> HandleRequestAsync(LSP.CompletionItem comp
 
         private static bool MatchesLSPCompletionItem(LSP.CompletionItem lspCompletionItem, CompletionItem completionItem)
         {
+            // We want to make sure we are resolving the same unimported item in case we have multiple with same name
```

Comment:
Instead of doing this, we could consider adding information to the completion item resolve data that would definitively indicate which one we're resolving.  Fine whichever you think is best.

Reply:
That does have the advantage of immune to change client might make to the resolve request in the future, with the cost of extra data on each unimported item. Let's take this fix for now, assuming such change would be very rare, but if it turns out not to be the case, we can switch to your proposal.